src/Payload/Server.purs

```diff
@@ -166,29 +179,54 @@ mkLogger logLevel = { log: log_, logDebug, logError }
     logError | logLevel >= LogError = log
     logError = const $ pure unit
 
-handleRequest :: Config -> Trie HandlerEntry -> HTTP.Request -> HTTP.Response -> Effect Unit
-handleRequest cfg@{ logger } routerTrie req res = do
+handleRequest
+  :: forall m
+   . MonadEffect m
+  => (m ~> Aff)
+  -> Config
+  -> Trie (HandlerEntry m)
+  -> HTTP.Request
+  -> HTTP.Response
+  -> Effect Unit
+handleRequest runM cfg@{ logger } routerTrie req res = do
   let url = Url.parse (HTTP.requestURL req)
   logger.logDebug (HTTP.requestMethod req <> " " <> show (url.path))
   case requestUrl req of
-    Right reqUrl -> runHandlers cfg routerTrie reqUrl req res
+    Right reqUrl -> runHandlers runM cfg routerTrie reqUrl req res
     Left err -> do
       writeResponse res (internalError $ StringBody $ "Path could not be decoded: " <> show err)
 
-runHandlers :: Config -> Trie HandlerEntry -> RequestUrl
-               -> HTTP.Request -> HTTP.Response -> Effect Unit
-runHandlers { logger } routerTrie reqUrl req res = do
-  let (matches :: List HandlerEntry) = Trie.lookup (reqUrl.method : reqUrl.path) routerTrie
+runHandlers
+  :: forall m
+   . MonadEffect m
+  => (m ~> Aff)
+  -> Config
+  -> Trie (HandlerEntry m)
+  -> RequestUrl
+  -> HTTP.Request
+  -> HTTP.Response
+  -> Effect Unit
+runHandlers runM { logger } routerTrie reqUrl req res = do
+  let (matches :: List (HandlerEntry m)) = Trie.lookup (reqUrl.method : reqUrl.path) routerTrie
   let matchesStr = String.joinWith "\n" (Array.fromFoldable $ (showRouteUrl <<< _.route) <$> matches)
   logger.logDebug $ showUrl reqUrl <> " -> " <> show (List.length matches) <> " matches:\n" <> matchesStr
-  Aff.launchAff_ $ do
-    outcome <- handleNext Nothing matches
-    case outcome of
-      (Forward msg) -> do
-        liftEffect $ writeResponse res (Response.notFound (StringBody ""))
-      _ -> pure unit
+  Aff.runAff_
+    (
+      case _ of
+        Left e -> liftEffect do
+          logger.logError $ message e
+          writeResponse res (internalError (StringBody ""))
+        _ ->
+          pure unit
+    )
```

Comment:
I found the change from `launchAff_` to `runAff_` to be necessary in case `runM` throws an error. The practical difference is that `runAff_` allows errors to be handled where `launchAff_` results in the server crashing.